app/src/main/java/com/example/weather/ui/forecast/ForecastFragment.java

```diff
@@ -64,12 +66,14 @@ public void onClick(View view) {
         viewModel.getLiveData().observe(this, new Observer<List<WeatherData>>() {
             @Override
             public void onChanged(List<WeatherData> weatherData) {
-                setAdapter(binding.recycler,weatherData);
+                setAdapter(binding.recycler,weatherData, viewModel);
```

Comment:
We don't need to set adapter every time on data changed. Better to set adapter once and then update data inside adapter.